addons/website_sale/migrations/9.0.1.0/post-migration.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# © 2016 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+from openerp import api, SUPERUSER_ID
+
+
+def set_last_sale_on_partner(env):
+    partners = env['res.partner'].search([])
```

Comment:
This is going to be awfully slow. You can use an SQL to make this:
```
UPDATE res_partner rp
SET last_website_so_id = so.id
FROM sale_order so
WHERE so.id = (
   SELECT max(id) FROM sale_order
   WHERE partner_id = rp.id
)
AND rp.last_website_so_id IS NULL;
```
Note that `website_order_line` is a one2many field that behaves the same of normal order_line (so all sales orders are website applicable).